source/initial_conditions/harmonic_perturbation.cc

```diff
@@ -35,12 +35,15 @@ namespace aspect
     HarmonicPerturbation<dim>::
     initial_temperature (const Point<dim> &position) const
     {
+      AssertThrow (this->has_boundary_temperature() || (!use_conductive_profile),
```

Comment:
At some point in the future move all of the changes to harmonic_perturbation to its own pull request. I will review it separately there.